packages/app/src/components/TxMonitoring.tsx

```diff
@@ -208,12 +210,38 @@ export function TxStatusList() {
   const { pendingTxs } = useContext(TxContext);
   const { chainId, account, active } = useWeb3React();
   const [pendingTxsState, pendingTxsDispatch] = pendingTxs;
-  const entries = Object.entries(pendingTxsState)
+  let entries = Object.entries(pendingTxsState)
     .sort((a, b) => b[1].created.getTime() - a[1].created.getTime())
     //  Only retain the most recent 10 records
     .slice(0, 10);
   const list: React.ReactElement[] = [];
 
+  if (Object.keys(entries).length !== 0) {
+    if (
+      localStorage.getItem("txAccount") &&
+      localStorage.getItem("txAccount") == account
+    ) {
+      localStorage.removeItem("txHistory");
+      localStorage.removeItem("txAccount");
+      localStorage.setItem("txAccount", JSON.stringify(account));
+      localStorage.setItem("txHistory", JSON.stringify(entries));
+    } else {
+      localStorage.removeItem("txHistory");
+      localStorage.removeItem("txAccount");
+      localStorage.setItem("txAccount", JSON.stringify(account));
+      localStorage.setItem("txHistory", JSON.stringify(entries));
+    }
+  }
+
+  if (
+    Object.keys(entries).length === 0 &&
```

Comment:
I think that what is there in local storage should alway be included in what is displayed - not only when there is nothing in pendingTxState.